File: InsertionSort.cpp
#include<iostream>
using namespace std;
void InsertionSort(int *a,int n){
    for(int i=1;i<=n;i++){
        int t=a[i];
        for(int j=i-1;j>=1;j--){
            if(a[j]>t){
                a[j+1]=a[j];
                a[j]=t;
            }
        }
    }
}
int main(){
    int n;
    cin>>n;
    int a[n+1];
    for(int i=1;i<=n;i++)
        cin>>a[i];
    InsertionSort(a,n);
    for(int i=1;i<=n;i++)
        cout<<a[i]<<' ';
    return 0;
    // 
}

File: InsertionSort.go
package main
import (
	"fmt"
)
func InsertionSort(a []int,n int){
	for i,val := range a{
		t := val
		for j:=i-1;j>=0;j--{
			if a[j]>t{
				a[j+1]=a[j]
				a[j]=t
			}
		}
	}
}
func main(){
	var a []int
	var n,x int
	fmt.Scanln(&n)
	for i:=0;i<n;i++{
		fmt.Scan(&x)
		a = append(a,x)
	}
	InsertionSort(a,n)
	fmt.Println(a)
	//
}